homestar-runtime/src/settings.rs

```diff
@@ -22,6 +24,44 @@ pub struct Monitoring {
     process_collector_interval: u64,
 }
 
+#[derive(Default, Clone, Debug, Deserialize)]
+/// Configure how the Network keypair is generated or using an existing one
+pub(crate) enum PubkeyConfig {
+    #[default]
+    Random,
+    /// Seed bytes are hashed with SHA-256 to produce the ed25519 secret key.
+    RandomWithSeed(Vec<u8>),
+    Existing(Vec<u8>),
```

Comment:
Though internally we can treat this as `Vec<u8>` @mriise, as far as configuration goes, should this come from a file path or something similar? Typically, the config will be pass in as toml/yaml, etc, and bytes won't be listed directly.

Reply:
Yep! this is changed in the newest commit!